docs/DeveloperGuide.md

```diff
@@ -1,38 +1,235 @@
+
+---
+### UI Component
+The UI is made up of three classes, `GeneralUi`, `CalorieUi` and `WeightUi`, and together,
+they improve on the accessibility of the application.
+
+The `UI` component,
+- Displays what are the possible inputs at any one time
+- Allows the user to check that a value added is correct
+- Informs the user whenever there is an invalid input
+
+### Storage Component
+
+![storage-class-diagram](./uml/storage-class-diagram.png)
+
+The `Storage` is the base class which all `Storage` components inherit from. There are currently four storages:
+
+1. `FoodStorage`: For parsing and storing all of the foods supported by _LifeTracker_
+2. `UserStorage`: For storing the user's profile
+3. `MealStorage`: For storing the meals that the user ate
+4. `ExerciseStorage`: For storing the exercises that the user did
+
+#### Interfaces
+
+To ensure that each `Storage` component only implements methods for writing and reading when needed, we created two interfaces
+
+1. `FileReadable`: Has `load()` method for `Storage` component to be able to read from a database
+2. `FileWritable`: Has `write()` method for `Storage` component to be able to write to a database
+
+For example, `FoodStorage` is not meant to be edited by user, hence it does not implement `FileWritable`
+
+# Implementation
+This section describes some noteworthy details on how certain features are implemented.  
+<p>&nbsp;</p>
+
+## [Proposed] Add meal feature
+
+### Proposed Implementation
+
+The proposed mechanism for adding a meal is facilitated by `AddMealCommand`. It extends `Command` and overrides the
+`execute` method in the `Command` class.
+
+In this command, there are 2 ways for the user to add a meal to storage. 
+1. Using a one-line command in their CLI in the format `add [DATE] /on [MEAL_TYPE] /type [FOOD_INDEX] /foods`
+2. Typing `add` into the CLI and following the printed prompts to enter the date in the specified format, type of meal and specific food.
+
+<img src="uml/AddMealCommand1.png" alt="Sequence Diagram" width="500">
+
+Step 1. As seen from the sequence diagram above, when the AddMealCommand is executed via the `execute` method in 
+LifeTracker, the user's input is first parsed to determine how he/she wants to input it. Either method sets the food, 
+date and meal type features necessary to create a new meal.
+
+Step 2. The constructor for the `Meal` class is called which instantiates a new instance of Meal using the
+parameters provided.
+
+Step 3. `mealStorage` saves the meal to the database and then `ui` prints out the confirmation of the meal added.
+
+## [Proposed] Delete meal feature
+
+### Proposed Implementation
+
+In order to delete a meal from their meal history, a user has to type `delete [MEAL_INDEX]` into the CLI. Users can see 
+which meal they want to delete by viewing the mealData.csv file.
 
-{Describe the target user profile}
+<img src="uml/DeleteMealCommand1.png" alt="Sequence Diagram" width="500">
```

Comment:
Is it possible to add the deletion of a Meal Object?
![image](https://user-images.githubusercontent.com/88478010/228723805-d089f378-e185-4a0a-810b-53584927695e.png)